resources/scripts/openapi-icon-tool/src/main/java/IconTool.java

```diff
@@ -0,0 +1,36 @@
+import java.io.File;
+import java.io.IOException;
+import java.util.List;
+
+public class IconTool {
+    static String iconsSourcePath;
+    static String openApiConnectorsPath;
+    public static void main(String[] args) throws IOException {
+        iconsSourcePath = Utils.getConfigValue("iconsSourcePath");
+        openApiConnectorsPath = Utils.getConfigValue("openApiConnectorsPath");
+        List<String> ballerinaPackages = Utils.findBallerinaPackages(openApiConnectorsPath);
+        for (String packagePath : ballerinaPackages) {
+            String[] packageNameSplit = packagePath.split("/");
+            String moduleName = packageNameSplit[packageNameSplit.length - 1];
+            boolean resourcesFolderExist = Utils.isResourcesFolderExist(packagePath);
+            if (resourcesFolderExist) {
+                String iconPath = Utils.findConnectorIcon(moduleName, iconsSourcePath);
+                if (iconPath != null) {
+                    Utils.addReplaceIcon(packagePath, iconPath, moduleName);
+                } else {
+                    System.out.println("No icon available for " + packagePath);
+                }
+            } else {
```

Comment:
This if and else has common stuff. What we do additionally is creating resources folder. What we can do is

```
createResoucesFolderIfNotExists(path or whatever params);
addNewIcon(packagePath, iconPath, moduleName);

private void  addNewIcon (String packagePath, String iconPath, String moduleName) {
  if (iconPath != null) {
                      Utils.addReplaceIcon(packagePath, iconPath, moduleName);
  } else {
                      System.out.println("No icon available for " + packagePath);
  }
}
```

Reply:
WDYT?

Reply:
Ack